src/core/aio/file_io.cpp

```diff
@@ -52,11 +73,20 @@ namespace file {
     auto cb = create_aio_task(callback_code, tracker, std::move(callback), hash);
     cb->get_aio_context()->buffer = buffer;
     cb->get_aio_context()->buffer_size = count;
-    cb->get_aio_context()->file = file;
+    cb->get_aio_context()->file_object = file;
+    cb->get_aio_context()->file = file->native_handle();
     cb->get_aio_context()->file_offset = offset;
     cb->get_aio_context()->type = AIO_Read;
+    cb->get_aio_context()->engine = &disk_engine::instance();
 
-    disk_engine::instance().read(cb);
+    if (!cb->spec().on_aio_call.execute(task::get_current_task(), cb, true)) {
+        cb->enqueue(ERR_FILE_OPERATION_FAILED, 0);
+        return cb;
+    }
+    auto wk = file->read(cb);
+    if (wk) {
+        disk_engine::provider()->submit_aio_task(wk);
+    }
```

Comment:
I don't see the `aio_task_ptr write` was refactored, is it no need?

Reply:
Because we have `file::write` and `file::write_vector` both using `disk_engine::write`. The write-process is much more complicated than others.  I would try this refactoring later.